RecoLocalMuon/DTRecHit/plugins/DTLinearDriftFromDBAlgo.cc

```diff
@@ -34,19 +36,37 @@ DTLinearDriftFromDBAlgo::DTLinearDriftFromDBAlgo(const ParameterSet& config)
       // Option to force going back to digi time at Step 2
       stepTwoFromDigi(config.getParameter<bool>("stepTwoFromDigi")),
       useUncertDB(config.getParameter<bool>("useUncertDB")),
+      readLegacyTTrigDB(config.existsAs<bool>("readLegacyTTrigDB") ? config.getParameter<bool>("readLegacyTTrigDB")
```

Comment:
::exists method is rather an exception left for cases for rather old and complex pre-7X software.
Define a ::fillDescriptions method https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp
If a fillDescriptions implementation is too complicated due to the tools/plugins coding pattern, please modify the existing configs appropriately and set the value to true/false there.

Reply:
I discussed this with @battibass before PR-ing and our opinion is that ,oving to fillDescriptions  consistently  would be too complicated since his module is re-used in several places upstream, cf:
https://github.com/cms-sw/cmssw/search?l=Python&q=DTLinearDriftFromDBAlgo

OTHO, all existing configs have already been modified in this same PR. The reason we added existsAs is not because of cfgs, but because this module is used in HLT and we want to allow for a smooth transition while the new cards are added to confDB.

Reply:
For HLT, a change can be inserted in https://github.com/cms-sw/cmssw/blob/master/HLTrigger/Configuration/python/customizeHLTforCMSSW.py

Reply:
Uhm the clean solution is obviously to update confDB which (in my understanding) is trivial once the card is present in the relevant cfi.py. At that point the temporary existsAs can be removed. I think that editing customizeHLTforCMSSW.py would be an equally temporary solution, and it seem less clean potentially and more error-prone.. 
What about open an issue to re-check and clean-up existAs in DT code instead, so that we don't forget to remove the temporary existAs()? (I think there are a few to be removed elsewhere as well)

Reply:
customizeHLTforCMSSW.py has been the preferred solution since a while

Reply:
I see but as I mentioned this module is reused in a number of places, cf:
https://github.com/cms-sw/cmssw/search?l=Python&q=DTLinearDriftFromDBAlgo
which means we would have to add quite a number of temporary changes in customizeHLTforCMSSW.py instead than a single line in the code... 
OTOH we have other existsAs to cleanup in the subsystem, so my proposal of coupling this with an issue would also go in the direction of a full cleanup and sync.
If that is not fine with you, my preference is to postpone modification of this class to a future PR, after the new cards have been added to confDB manually (and proceed with the rest of the PR now).

Reply:
>  If that is not fine with you, my preference is to postpone modification of this class to a future PR, after the new cards have been added to confDB manually (and proceed with the rest of the PR now).

My preference in the current context (no dillDescriptions) is to have a customizeHLTforCMSSW.py and no `existsAs`

Reply:
I understand but I really have a hard time figuring out how this would be done.
The new cards should be added to these psets:

- hltDt1DRecHits.recAlgoConfig
- hltDt4DSegments.Reco4DAlgoConfig.recAlgoConfig
- hltDt4DSegments.Reco4DAlgoConfig.Reco2DAlgoConfig.recAlgoConfig
- hltDt4DSegmentsMeanTimer.Reco4DAlgoConfig.recAlgoConfig
- hltDt4DSegmentsMeanTimer.Reco4DAlgoConfig.Reco2DAlgoConfig.recAlgoConfig

How should that be coded in customizeHLTforCMSSW.py?
is there a way to add only the new cards, or should the entire pset be replaced?

Reply:
there are relevant examples in customizeHLTforCMSSW.py already for replacement/parameter insertions by the module type.

IIUC, `DTRecHitProducer` is the first kind
```python
for producer in producers_by_type(process, "DTRecHitProducer"):
  producer.recAlgoConfig.readLegacyTTrigDB = cms.bool(False)
```
the other cases are apparently `DTRecSegment4DProducer`

Reply:
Should be done now. As far as I can tell, HLT runs correctly with the latest commit.